src/discordtipbot/coin_api.cr

```diff
@@ -45,10 +45,9 @@ class CoinApi
   end
 
   def balance : BigDecimal
-    info = get_info
-    raise "There was an error getting the balance of the node" unless info.is_a?(Hash(String, JSON::Type))
+    bal = @rpc.get_balance("*", 0)
 
-    BigDecimal.new(info["balance"].to_s) || BigDecimal.new("0")
+    BigDecimal.new(bal.to_s) || BigDecimal.new("0")
```

Comment:
We should be using [`BigDecimal#initialize(value : Int32)`](https://carc.in/#/r/3iku) instead of tossing a string away.

Reply:
It's not an Int32 though. IIRC it can be any type in theory.

Reply:
I'm talking about `BigDecimal.new("0")`.

Reply:
Oh. Sure. What advantage does it give? Avoiding allocation of the string?

Reply:
Yes